aws-yaml-ansible-wordpress/Pulumi.yaml

```diff
@@ -0,0 +1,243 @@
+name: pulumi-ansible-wordpress
+runtime: yaml
+description: Deploy an EC2 Wordpress server using Pulumi and Ansible.
+
+configuration:
+  publicKeyPath:
+    type: String
+    required: true
+  privateKeyPath:
+    type: String
+    required: true
+  dbInstanceSize:
+    type: String
+    default: db.t3.small
+  dbName:
+    type: String
+    default: wordpressdb
+  dbUsername:
+    type: String
+    default: admin
+  dbPassword:
+    type: String
+    required: true
+  ec2InstanceSize:
+    type: String
+    default: t3.small
+
+variables:
+  # Dynamically fetch AZs so we can spread across them.
+  availabilityZones:
+    Fn::Invoke:
+      Function: aws:getAvailabilityZones
+  # Dynamically query for the Amazon Linux 2 AMI in this region.
+  awsLinuxAmi:
+    Fn::Invoke:
+      Function: aws:ec2:getAmi
+      Arguments:
+        owners: ["amazon"]
+        filters:
+          - name: name
+            values: ["amzn2-ami-hvm-*-x86_64-ebs"]
+        mostRecent: true
+  # Read in the public key for easy use below.
+  publicKey:
+    Fn::ReadFile: ${publicKeyPath}
+  # Read in the private key for easy use below (and to ensure it's marked a secret!)
+  privateKey:
+    Fn::Secret:
+      Fn::ReadFile: ${privateKeyPath}
+
+resources:
+  # Set up a Virtual Private Cloud to deploy our EC2 instance and RDS datbase into.
+  prod-vpc:
```

Comment:
Suggest using AWSX 1.0 for this if it works with YAML (I forget if it does).

Reply:
Thanks Josh. I added a note about the fact that you can choose to use AWSX. Especially for the Ansible audience, I actually thought it would be best to lay out all the raw building blocks so there's a bit less "magic" to the example.